ext/ddtrace_profiling_native_extension/ruby_helpers.h

```diff
@@ -57,3 +57,9 @@ NORETURN(void raise_unexpected_type(
   int line,
   const char* function_name
 ));
+
+// This API is exported as a public symbol by the VM BUT the function header is not defined in any public header, so we
+// repeat it here to be able to use in our code.
+//
+// Queries if the current thread is the owner of the global VM lock.
+int ruby_thread_has_gvl_p(void);
```

Comment:
Looks useful to CRuby upstream 🤔

Reply:
Yeah... To be honest sometimes with the exposed APIs it's kinda unclear if something was incredibly deliberate or just an accident that happened.

I still have a few more things I want to annoy upstream about first, so I'll leave reporting this for a later time. (And it's quite possible they'll just remove this API in the near future, I believe ko1 is working on M:N threads support, aka not directly mapping Ruby threads 1:1 to operating system threads)